### PR TITLE
Rsa support

### PR DESCRIPTION
**Pull request by darkgl0w**
Address this issue: https://github.com/fastify/fastify-jwt/issues/25#issue-370672925

Refactored functions and code organization, swapped `var` with `let` in order to "isolate" variables inside the plugin, renamed some variables to follow [jsonwebtoken](https://github.com/auth0/node-jsonwebtoken) package variables naming conventions ... etc.

That's not perfect yet, but for now it does the job (I will complete this work later with another PR).

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -291,7 +291,7 @@ test('sign and verify with RSA and options', function (t) {
     t.plan(2)
 
     const fastify = Fastify()
-    fastify.register(jwt, { secret: 'test' })
+    fastify.register(jwt, { secret: { key: privateKey, passphrase: publicKey }, options: { issuer: 'Some issuer', subject: 'Some subject', audience: 'Some audience', algorithm: 'RS256' } })
```
Comment (mcollina) [inline test.js]: can you split this on multiple lines?

**Comment by darkgl0w** [comment]
Done, I refactored all the single lines to split them on multiple lines (I hope it's OK as that makes reading the code better). ^^

**Comment by SkeLLLa** [comment]
Probably it also worth to mention that besides `key` and `passphrase` in private key section
```
private: {
      key: readFileSync(`${path.join(__dirname, 'certs')}/private.pem`),
      passphrase: 'super secret passphrase'
    },
```
there could also be `cipher` parameter that controls (en)decryption algorithm of private key. This could be handy when private key is generated without headers like `DEK-Info:` and so on. For example, node 10.12 `crypto.generateKeyPair` produces key without header and it could fail on token decryption if no cipher provided.

**Comment by darkgl0w** [comment]
@SkeLLLa > yes that's interesting, currently I just tried to make fastify-jwt and jsonwebtoken as close as possible and to me that's not perfect yet, in its current state the package can now manage to use even ECDSA keys but I need to write some verifications to avoid any unwanted behavior, polish error messages, implement a better options management ... and write more tests to cover even more use cases... yeah in case you ask I'm not yet satisfied :'D.

About the `cipher` option key I didn't saw any reference to this in jsonwebtoken code (I will check back tomorrow to be sure), but if I am not mistaken, we should first add it to jsonwebtoken (I guess this would be an interesting feature PR to work for both jsonwebtoken and fastify-jwt packages). :P

**Comment by SkeLLLa** [comment]
@darkgl0w I use jsonwebtoken and today with newly introduced nodejs 10.12 crypto keypair generation. I will try to isolate issue with examples.

**Comment by darkgl0w** [comment]
Hi,
with those last commits I think it's now good. :rocket: 

We can now use RSA and ECDSA certs (with and without encryption) without any unwanted side effects. I tried to test all the use cases I had in mind :d

I have a few more changes in mind (options management in particular) but I need to polish the reflection behind this refactor to make something clean that does not introduce any breaking change so I will do it in a separate PR.

@SkeLLLa > ok that's nice :D thank you for taking the time to do some tests. ^^

**Comment by mcollina** [comment]
Thanks so much for working on this, highly appreciated.

**Pull request merged**